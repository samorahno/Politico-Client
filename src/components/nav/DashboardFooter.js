import React from 'react';
import '../../styles/dashboard.css';

const Footer = () => {
    return (
        <div>
            <footer>
		        &copy; 2019 Politico by Abosede Samson
	        </footer>
        </div>
    );
};

export default Footer;
