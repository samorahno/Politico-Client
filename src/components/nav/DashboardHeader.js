import React from 'react';
import '../../styles/dashboard.css';

const Header = () => {
    return (
        <div>
            <header>
		        <a href="dashboard.html"><span />Politico</a>

                <a href="#"
                    className="minimize_button"
                    onClick="togle_nav_func()"
                >
                    <span><i className="fa fa-bars fa-1x" /></span></a>
	        </header>
        </div>
    );
};

export default Header;
