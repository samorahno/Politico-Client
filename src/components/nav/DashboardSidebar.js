import React from 'react';
import { NavLink, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import '../../styles/dashboard.css';
import { logoutUser } from '../../actions';

export const Sidebar = (props) => {
    const { logout, history } = props;
    const logoutSuccess = () => {
        logout();
        history.push('/');
    };
    return (
        <div className="side-nav visible" id="togle_nav">
            <nav>
                <ul>
                    <li>
                        <NavLink
                            to="/dashboard"
                            activeClassName="active"
                        >
                            <span > <i className="fa fa-home fa-fw" /> </span>
                            <span>Home</span>
                        </NavLink>
                    </li>
                    <li>
                        <NavLink to="/party">
                            <span > <i className="fa fa-home fa-fw" /> </span>
                            <span>Political Party</span>
                        </NavLink>
                    </li>
                    <li>
                        <NavLink to="/office">
                            <span > <i className="fa fa-compress" /> </span>
                            <span>Political Office</span>
                        </NavLink>
                    </li>
                    <li>
                        <NavLink to="/candidate">
                            <span > <i className="fa fa-pencil-square" /> </span>
                            <span>Candidate</span>
                        </NavLink>
                    </li>

                    <li>
                        <NavLink to="/vote">
                            <span > <i className="fa fa-envelope" /> </span>
                            <span>Vote</span>
                        </NavLink>
                    </li>

                    <li>
                        <NavLink to="/result">
                            <span > <i className="fa fa-envelope" /> </span>
                            <span>Vote Result</span>
                        </NavLink>
                    </li>

                    <li>
                        <NavLink to="/profile">
                            <span > <i className="fa fa-pencil" /> </span>
                            <span>View Profile</span>
                        </NavLink>
                    </li>
                    <li style={{ color: '#fff', cursor: 'pointer' }} onClick={logoutSuccess}>
                        <span> <i className="fa fa-power-off" /> </span>
                        <span>Logout</span>
                    </li>

                </ul>
            </nav>
        </div>
    );
};

const mapDispatchToProps = ({
    logout: logoutUser
});

export default withRouter(connect(null, mapDispatchToProps)(Sidebar));
