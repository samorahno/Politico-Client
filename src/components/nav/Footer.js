import React from 'react';
import '../../styles/landing.scss';

const Footer = () => {
    return (
        <div>
            <footer className="steps footercolor">
                <div className="column step">
                    <table className="footer-table">
                        <tbody>
                            <tr>
                                <td><span className="footer-td">
                                    <p><i className="fa fa-phone" /> : +2348060955366 <br /> <i className="fa fa-map-marker" /> : 28,
                                        Adewale Street, Victoria Island, Lagos, Nigeria <br /><i className="fa fa-envelope" /> :
                                        samabos007@gmail.com
                                    </p>
                                </span>
                                </td>
                            </tr>
                        </tbody>
                    </table>
                </div>
                <div className="column step">
                    <p>
                        Politico | by Abosede Samson &copy; 2019
                    </p>
                    <p />
                </div>
            </footer>
        </div>
    );
};

export default Footer;
