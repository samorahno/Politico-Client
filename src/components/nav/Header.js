import React from 'react';
import '../../styles/landing.css';
const Header = () => {
    return (
        <div>
            <div className="header1">
                <a href="index.html">Politico</a>
            </div>
        </div>
    );
};

export default Header;
