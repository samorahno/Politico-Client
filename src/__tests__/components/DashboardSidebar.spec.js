import React from 'react';
import { shallow } from 'enzyme';
import { Sidebar } from '../../components/nav/DashboardSidebar';

const props = {
    logout: jest.fn(),
    history: {
        push: jest.fn()
    },
    logoutSuccess: jest.fn()
};

describe('Dashboard Sidebar', () => {
    it('should render dashboard sidebar component correctly', () => {
        const component = shallow(<Sidebar {...props} />);
        const logoutSuccess = jest.fn();
        expect(logoutSuccess.mock.calls.length).toBe(0);
        expect(component).toMatchSnapshot();
    });
});
