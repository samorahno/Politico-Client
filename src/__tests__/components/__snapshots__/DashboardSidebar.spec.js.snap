// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Sidebar should render dashboard sidebar component correctly 1`] = `ShallowWrapper {}`;
